Doc/glossary.rst

```diff
@@ -289,6 +299,23 @@ Glossary
       advanced mathematical feature.  If you're not aware of a need for them,
       it's almost certain you can safely ignore them.
 
+   concurrency
+      The ability of different parts of a program to be executed out-of-order
+      or in partial order without affecting the outcome.  This allows for
+      multiple tasks to make progress during overlapping time periods, though
+      not necessarily simultaneously.  In Python, concurrency can be achieved
+      through :mod:`threading` (using OS threads), :mod:`asyncio` (cooperative
+      multitasking), or :mod:`multiprocessing` (separate processes).
+      See also :term:`parallelism`.
+
+   concurrent modification
+      When multiple threads modify shared data at the same time without
+      proper synchronization.  Concurrent modification can lead to
+      :term:`data races <data race>` and corrupted data.  In the
+      :term:`free-threaded <free threading>` build, built-in types like
+      :class:`dict`, :class:`list`, and :class:`set` use internal locks
+      to protect against concurrent modifications.
```

Comment:
Maybe it makes sense to talk about third-party extensions that might not necessarily prevent data races?